lib/Dialect/SV/SVOps.cpp

```diff
@@ -244,16 +244,37 @@ LogicalResult LocalParamOp::verify() {
 // RegOp
 //===----------------------------------------------------------------------===//
 
+static ParseResult
+parseImplicitInitType(OpAsmParser &p, mlir::Type regType,
+                      std::optional<OpAsmParser::UnresolvedOperand> &initValue,
+                      mlir::Type &initType) {
+  if (!initValue.has_value())
+    return success();
+
+  hw::InOutType ioType = regType.dyn_cast<hw::InOutType>();
+  if (!ioType)
```

Comment:
This seems like it should be checked elsewhere, so this check is redundant.

Reply:
Can you just do a cast, or has the other check not run yet?

Reply:
I added a check here since this is done during parse time. I'm fairly confident that the (type) verifiers for an op will only run after the op has been parsed successfully.

Reply:
Can't you just constrain the init type to the stripped operation type in ods?  Then you don't need any of this.